libcanard/canard_dsdl.c

```diff
@@ -31,8 +31,15 @@
 #    define CANARD_PRIVATE static
 #endif
 
-#if !defined(__STDC_VERSION__) || (__STDC_VERSION__ < 201112L)
-#    error "Unsupported language: ISO C11 or a newer version is required."
+#if !defined(__STDC_VERSION__) || (__STDC_VERSION__ < 199901L)
+#    error "Unsupported language: ISO C99 or a newer version is required."
+#endif
+
+/// Flag for masking the static assertions when building with compilers prior to C11
+#if (__STDC_VERSION__ < 201112L) && !defined(CANARD_ALLOW_STATIC_ASSERTION)
+    #define CANARD_ALLOW_STATIC_ASSERTION  false
+#else
+    #define CANARD_ALLOW_STATIC_ASSERTION  true
 #endif
```

Comment:
This logic adds too much boilerplate to static assertion checks. Please check if [`static_assert`](https://en.cppreference.com/w/c/error/static_assert) is defined, and if not, use the traditional negative-sized array declaration macro. See o1heap as example.